open_instruct/test_utils.py

```diff
@@ -75,6 +76,26 @@ def test_time_parser_used_in_get_beaker_dataset_ids(self):
         self.assertTrue(parser.parse("0001-01-01T00:00:00Z"))
 
 
+class TestUtilityFunctions(unittest.TestCase):
+    """Test utility functions in utils module."""
+
+    @parameterized.expand(
+        [
+            ("basic_repeat", ["a", "b", "c"], 3, ["a", "a", "a", "b", "b", "b", "c", "c", "c"]),
+            ("repeat_once", ["a", "b", "c"], 1, ["a", "b", "c"]),
+            ("repeat_zero", ["a", "b", "c"], 0, []),
+            ("empty_sequence", [], 3, []),
+            ("integers", [1, 2, 3], 2, [1, 1, 2, 2, 3, 3]),
+            ("mixed_types", ["a", 1, None, True], 2, ["a", "a", 1, 1, None, None, True, True]),
+            ("single_element", ["x"], 5, ["x", "x", "x", "x", "x"]),
+        ]
+    )
+    def test_repeat_each(self, name, sequence, k, expected):
+        """Test the repeat_each function with various inputs."""
+        result = repeat_each(sequence, k)
+        self.assertEqual(result, expected)
```

Comment:
really minor nit would be that I'd wanna test if mutating a sequence item after repeat each doesn't change the repeated versions, but this is really minor.

Reply:
Done.